lib/Dialect/XTenNN/IR/XTenNNOps.cpp

```diff
@@ -215,15 +223,90 @@ static void printKernelArgumentList(OpAsmPrinter &p, TypeRange types,
 }
 
 // Parse
-//  $name custom<KernelArgumentList>(type($arguments), $arguments) attr-dict
+// {((name = )?value, )*((name = )?value)}
+static ParseResult parseKernelInstantiationArgs(OpAsmParser &p,
+                                                SmallVector<Attribute> &values,
+                                                SmallVector<Attribute> &names) {
+  if (failed(p.parseLBrace()))
+    return failure();
+
+  if (failed(p.parseCommaSeparatedList([&p, &names, &values]() {
+        std::string name;
+        bool hasName = false;
+        if (succeeded(p.parseOptionalKeywordOrString(&name))) {
```

Comment:
From reading the doc, `parseOptionalKeywordOrString` seems to either expect a keyword
https://github.com/xilinx/llvm-project/blob/3db9abb906f344e400fe106943ce7e24df3911b6/mlir/lib/AsmParser/TokenKinds.def#L83
or a quoted string (like `"mystring"`),
I don't understand how this is able to parse `N = 42 : i32`, where `N` is a `bare_identifier` from what I understand.
(At least that is how the DictAttribute parsing works: https://github.com/xilinx/llvm-project/blob/3db9abb906f344e400fe106943ce7e24df3911b6/mlir/lib/AsmParser/AttributeParser.cpp#L306)

Reply:
_C'est un miracle !_
I changed this to expect a quoted string to lift this ambiguity, although I liked the quote-less identifiers.